Add security recommendations for microfrontends (#2783)
kyma-project/kyma: docs/console/docs/01-01-console.md

- The MicroFrontend custom resource allows you to plug in micro front-ends for a specific Namespace.
- The ClusterMicroFrontend custom resource allows you to plug in micro front-ends for the entire Cluster.
- The BackendModule custom resource allows you to enable UI API Layer modules.
  

> #### Security guidelines for `MicroFrontend` and `ClusterMicroFrontend` resources:

Reviewer: ```suggestion
# Security guidelines for MicroFrontend and ClusterMicroFrontend custom resources
```
Add this as the title of the file in the title metadata box.

> We highly recommend to include the following configuration to your `MicroFrontend` or `ClusterMicroFrontend` web server config file:

Reviewer: ```suggestion
For security reasons, follow the listed guidelines when you add the MicroFrontend or ClusterMicroFrontend configuration in the web server file:
```
I would say the following points are rather guidelines than the configuration itself, that's why I changed the introductory line - for the user to follow the guidelines when defining the configuration that goes into the web server file.

Author: `Web server file` sounds weird IMHO, and also this is unrelated to the resource configuration itself, but to the web server serving the microfrontend. This would be my counterproposal, let me know if it is fine for you:
_For security reasons, follow the listed guidelines when you configure the web server for the MicroFrontend or ClusterMicroFrontend_

Author: done

>  - Make it only accessible through HTTPS.

Reviewer: ```suggestion
 - Make the configuration accessible only through HTTPS.
```

Author: What about `Make the MicroFrontend or ClusterMicroFrontend accessible oly through HTTPS`?

Author: done

>  - HTTP header `Access-Control-Allow-Origin` should be as restrictive as possible.

Reviewer: ```suggestion
 - Make the **Access-Control-Allow-Origin** HTTP header as restrictive as possible.
```

>  - HTTP header `X-Content-Type` should be set to `nosniff`.

Reviewer: ```suggestion
 - Set the  **X-Content-Type HTTP** header to `nosniff`.
```

>  - HTTP header `X-Frame-Options` should be set to `sameorigin` or `allow-from ALLOWED_URL`.

Reviewer: ```suggestion
 - Set the **X-Frame-Options** HTTP header to `sameorigin` or `allow-from ALLOWED_URL`.
```
